components/merkledb/src/validation.rs

```diff
@@ -55,14 +55,14 @@ where
 {
     if name.as_ref().is_empty() {
         Err("Index name must not be empty".into())
-    } else if !predicate(name) {
+    } else if predicate(name) {
```

Comment:
Ok case between error cases looks weird.

Reply:
Yes, but clippy doesn't agree. I think it better to turn it off.

Reply:
maybe 
```rust
if name.as_ref().is_empty() {
        Err(...)
} else if !predicate(name) {
        Err(...)
} else {
        Ok(...)
}
```

it looks more natural.

Reply:
It was previous variant, I've changed it due to clippy complains. Will revert it back and turn off clippy.